plugins/notifiers/client.go

```diff
@@ -92,3 +178,129 @@ func NewSlackConfig(config *Config) (*slack.Config, error) {
+	// }
+
+	return slackConfig, nil
+}
+
+func GetLarkConfig(config *Notifier, messages domain.NotificationMessages) (*lark.Config, error) {
+	// validation
+	if config.ClientID == "" && config.ClientSecret == "" {
+		return nil, errors.New("lark clientid & clientSecret must be provided")
+	}
+	if config.ClientID == "" && config.ClientSecret != "" {
+		return nil, errors.New("lark clientid & clientSecret must be provided")
+	}
+	if config.ClientID != "" && config.ClientSecret == "" {
+		return nil, errors.New("lark clientid & clientSecret must be provided")
+	}
+
+	var larkConfig *lark.Config
+	if config.ClientID != "" {
+		workspaces := []lark.LarkWorkspace{
+			{
+				WorkspaceName: "default",
+				ClientId:      config.ClientID,
+				ClientSecret:  config.ClientSecret,
+				Criteria:      "1==1",
+			},
+		}
+		larkConfig = &lark.Config{
+			Workspaces: workspaces,
+			Messages:   messages,
+		}
+		return larkConfig, nil
+
+	}
+	// var workSpaceConfig lark.WorkSpaceConfig
+	// if err := config.LarkConfig.Decode(&workSpaceConfig); err != nil {
+	// 	return nil, fmt.Errorf("invalid lark workspace config: %w", err)
+	// }
+
+	// larkConfig = &lark.Config{
+	// 	Workspaces: workSpaceConfig.Workspaces,
+	// 	Messages:   config.Messages,
+	// }
+
+	return larkConfig, nil
+}
+
+// func NewMultiClientConfig(config *ConfigMultiClient) (*multiclient.Config, error) {
+// 	// validation
+// 	if config.Provider == "" {
+// 		return nil, errors.New("multiConfig must be provided")
+// 	}
+// 	var configg ConfigMultiClient
+// 	// Unmarshal YAML into map[string]interface{}
+// 	var c map[string]interface{}
+
+// 	// Use mapstructure to decode map into struct
+// 	errrr := mapstructure.Decode(c, &configg)
+// 	if errrr != nil {
+// 		//log.Fatalf("error decoding: %v", errrr)
+// 	}
+// 	fmt.Println("lark send config 3" + configg.ClientConfig + "multiConfig.Workspaces")
+// 	var multiConfig *multiclient.Config
+// 	if config.Provider != "" {
+// 		workspaces := []multiclient.Workspace{
+// 			{
+// 				WorkspaceName: "config.Decode()",
+// 				ClientId:      "config.Decode()",
+// 				Criteria:      "1==1",
+// 			},
+// 		}
+// 		multiConfig = &multiclient.Config{
+// 			Workspaces: workspaces,
+// 			Messages:   config.Messages,
+// 		}
+
+// 		for _, obj := range workspaces {
+// 			fmt.Println("lark send config 4 " + obj.ClientId)
+// 		}
+// 		return multiConfig, nil
+// 	}
+// 	fmt.Println("lark send config 3" + config.Provider + "multiConfig.Workspaces")
+// 	var workSpaceConfig multiclient.WorkSpaceConfig
+// 	if err := config.MultiConfig.Decode(&workSpaceConfig); err != nil {
+// 		return nil, fmt.Errorf("invalid slack workspace config: %w", err)
+// 	}
+
+// 	multiConfig = &multiclient.Config{
+// 		Workspaces: workSpaceConfig.Workspaces,
+// 		Messages:   config.Messages,
+// 	}
+// 	fmt.Println("lark send workspace 2 " + "multiConfig.Workspaces")
+// 	return multiConfig, nil
+// }
+
+func (nm *NotifyManager) AddClient(client Client) {
```

Comment:
we don't have to export (public) this method
```suggestion
func (nm *NotifyManager) addClient(client Client) {
```